src/threads.cpp

```diff
@@ -63,9 +73,11 @@ void *reading_thread(void *arg) {
 		aggIntEnd = time(NULL);
 		do { /* start thread main loop */
 			do {
+				_safe_to_cancel();
 				aggIntEnd += mtr->aggtime(); /* end of this aggregation period */
 			} while ((aggIntEnd < time(NULL)) && (mtr->aggtime() > 0));
 			do { /* aggregate loop */
+				_safe_to_cancel();
```

Comment:
how about the sleep 5 lines below?

Reply:
did you think about using deferred cancellation? sleep is a cancellation point so it would work for that.